### ISSUE TITLE
Support for maximum line length in freshmark and FormatExtension configurations 

### ISSUE THREAD

**Issue by JLLeitschuh** [diffplug/spotless#114]
It would be awesome if you could set a maximum line length for `.md` files and for build files.

**Comment by nedtwigg** [comment]
The way to do this would be to create a new FormatterStep that wraps all lines which are longer than an input parameter.

You can create such a FormatterStep in your buildscript, and it works well, we'd love a PR to bring it into Spotless core.

https://github.com/diffplug/spotless/blob/master/CONTRIBUTING.md

Here's a difficult user story:

- I limit my line length to 80 chars
- I change the limit to 60 chars
- Do I end up with line lengths
    - like this: `60 20 60 20`
    - or like this: `60 60 60 60`

A naive implementation will result in `60 20 60 20`.  In order to reflow line breaks and get the nice `60 60 60 60` result, you need to understand the underlying text well enough to know which line breaks are signifcant.

I'm not sure if the naive implementation is useful enough to be a good fit for spotless-lib, but maybe it is 👍

**Comment by JLLeitschuh** [comment]
I guess the line length for readme really could only be a failing check. The user would have to go figure out how to resolve.
This is probably counter to the mentality of the plugin.

**Comment by nedtwigg** [comment]
> This is probably counter to the mentality of the plugin.

This is true for spotless' builtins, but it's easy to use it in this way on your own:

https://github.com/diffplug/spotless/blob/823585cad2b51004836a0684ee7872ad53ca14b0/spotlessSelf.gradle#L11-L16

**Comment by JLLeitschuh** [comment]
Is `it` a single line or the entire document??

**Comment by jbduncan** [comment]
@JLLeitschuh `it` refers to an entire source file (or document, as you called it). :)
